pkgs/development/ocaml-modules/elpi/default.nix

```diff
@@ -1,21 +1,21 @@
 { lib, fetchzip, buildDunePackage, camlp5
-, ppx_tools_versioned, ppx_deriving, re
+, ppxlib, ppx_deriving, re, perl, ncurses
 }:
 
 buildDunePackage rec {
   pname = "elpi";
-  version = "1.11.2";
+  version = "1.11.4";
 
    src = fetchzip {
      url = "https://github.com/LPCIC/elpi/releases/download/v${version}/elpi-v${version}.tbz";
-     sha256 = "15hamy9ifr05kczadwh3yj2gmr12a9z1jwppmp5yrns0vykjbj76";
+     sha256 = "1hmjp2z52j17vwhhdkj45n9jx11jxkdg2dwa0n04yyw0qqy4m7c1";
    };
 
   minimumOCamlVersion = "4.04";
 
-  buildInputs = [ ppx_tools_versioned ];
+  buildInputs = [ perl ppxlib ];
 
-  propagatedBuildInputs = [ camlp5 ppx_deriving re ];
+  propagatedBuildInputs = [ ncurses camlp5 ppx_deriving re ];
```

Comment:
Why this change?

Reply:
According to @gares if I understood well `elpi` may need `tput` at runtime to display outputs in some context, and `ncurses` was the most sensible derivation exposing it... would some other changes make more sense?

Reply:
propagatedBuildInputs is not for runtime dependencies. You may either wrap the elpi binaries to set the PATH environment variable or hardcode the path to tput in the source code.

Reply:
@vbgl thanks! I do not know how to implement either of these solutions, can you give me some doc and/or examples?

Reply:
For instance, the opam binary is wrapped: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/ocaml/opam/default.nix#L104

Coq hardcodes the path to csdp: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/science/logic/coq/default.nix#L44

A third possibility is just to leave a comment: “elpi will assume your terminal is 80 positions wide, unless tput is found in PATH”.

Reply:
@vbgl, thanks a lot for you advice, I took the option to hardcode the path and it works well, thank you!

Reply:
You’re welcome. Thanks for your contributions.